test/oli_web/live/delivery/student/learn_live_test.exs

```diff
@@ -1994,6 +2017,603 @@ defmodule OliWeb.Delivery.Student.ContentLiveTest do
     end
   end
 
+  describe "student at Outline view mode" do
+    setup [:user_conn, :create_elixir_project]
+
+    test "can not access when not enrolled to course", %{conn: conn, section: section} do
+      {:error, {:redirect, %{to: redirect_path, flash: _flash_msg}}} =
+        live(conn, Utils.learn_live_path(section.slug, selected_view: :outline))
+
+      assert redirect_path == "/unauthorized"
+    end
+
+    test "can access when enrolled to course", %{conn: conn, user: user, section: section} do
+      Sections.enroll(user.id, section.id, [ContextRoles.get_role(:context_learner)])
+      Sections.mark_section_visited_for_student(section, user)
```

Comment:
One thing commonly seen in most of our tests is a piece of code repeatedly called at the beginning of all the test cases (I found myself doing it as well).
I think it could be cleaner to have one or maybe two functions such as `:enroll` and `:mark_section_visited` and call them on the `describe`'s setup.